src/components/picker/option/index.tsx

```diff
@@ -67,6 +67,11 @@ export class SmoothlyPickerOption {
 		if (!this.readonly && !this.required)
 			this.selected = !this.selected
 	}
+	@Method()
+	async setSelected(selected: boolean) {
```

Comment:
i am a little unsure how well this method will do in this component. This component goes through `document.cloneNode()` and in my testing clones nodes did not necessary have everything that was expected of them (which is why the madness on line 22-44 exists + the corresponding restore function exists in the menu component). We can also see that this already exists on line 37 so we now have 2 functions/methods that do the same thing. is it possible to rewrite it so that we only have the one on line 37?

Reply:
i stared myself blind on clickHandler() and totally missed this! line 37 does the trick!